Automattic/woocommerce-payments
# Clarify "disputed" order note: add amount and deadline

Fixes #6379 

#### Changes proposed in this Pull Request

This PR updates the order note added to disputed orders to make it more informative and more actionable. 

Example (single currency store):

<img width="427" alt="Screenshot 2023-05-31 at 1 18 32 PM" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/772bc515-ffef-40cc-840b-2ab237f73d1c">

Example (multi-currency store):

<img width="481" alt="Multicurrency store disputed order note" src="https://github.com/Automattic/woocommerce-payments/assets/4167300/a1ab3361-e5d1-4e2f-8c69-4df72b36aea2">


Previous message:

> Payment has been disputed as %1$s. See <a>dispute overview</a> for more details.

New message includes amount and deadline date: 

> Payment has been disputed for %1$s with reason %2$s. <a>Response due by %3$s</a>.

Note that the reason displayed is the raw value from Stripe API and is not currently converted to human-readable string. This will need to be updated, either as part of #6356 (see checklist item 2: `Ensure consistent naming of dispute types across WCPay platform (server, client, documentation)`) or a separate issue.

#### Testing instructions
The order note is generated in response to a webhook, so you will need webhooks working on your store. I did this by testing against WCPay production server and using [ngrok](https://ngrok.com/) so my dev site has a publicly available hostname, can also use WCPay dev tools and local WCPay server.

- Set up store so connected to WCPay (test mode) and have products to buy.
- Purchase a product using a [dispute card number](https://stripe.com/docs/testing#disputes) e.g. `4000000000000259`
- View the order edit page for the order
- Should show an informative dispute notice with correct details
  - Please crosscheck deadline, amount etc are correct and consistent with dispute details

-------------------


**Post merge**



- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : https://github.com/Automattic/woocommerce-payments/issues/6356#issuecomment-1571099446
